src/material-experimental/mdc-chips/chip-grid.ts

```diff
@@ -6,19 +6,418 @@
  * found in the LICENSE file at https://angular.io/license
  */
 
-import {ChangeDetectionStrategy, Component, ViewEncapsulation} from '@angular/core';
+import {coerceBooleanProperty} from '@angular/cdk/coercion';
+import {BACKSPACE} from '@angular/cdk/keycodes';
+import {
+  AfterContentInit,
+  AfterViewInit,
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  Component,
+  DoCheck,
+  ElementRef,
+  EventEmitter,
+  Input,
+  OnDestroy,
+  Optional,
+  Output,
+  Self,
+  ViewEncapsulation
+} from '@angular/core';
+import {ControlValueAccessor, FormGroupDirective, NgControl, NgForm} from '@angular/forms';
+import {
+  CanUpdateErrorState,
+  CanUpdateErrorStateCtor,
+  ErrorStateMatcher,
+  mixinErrorState,
+} from '@angular/material/core';
+import {MatFormFieldControl} from '@angular/material/form-field';
+import {MatChipTextControl} from './chip-text-control';
+import {merge, Observable, Subscription} from 'rxjs';
+import {startWith, takeUntil} from 'rxjs/operators';
 
+import {MatChipEvent} from './chip';
+import {MatChipSet} from './chip-set';
+
+
+/** Change event object that is emitted when the chip grid value has changed. */
+export class MatChipGridChange {
+  constructor(
+    /** Chip grid that emitted the event. */
+    public source: MatChipGrid,
+    /** Value of the chip grid when the event was emitted. */
+    public value: any) { }
+}
+
+// Boilerplate for applying mixins to MatChipGrid.
+/** INTERNAL USE ONLY. */
+class MatChipGridBase extends MatChipSet {
+  constructor(_elementRef: ElementRef,
+              _changeDetectorRef: ChangeDetectorRef,
+              public _defaultErrorStateMatcher: ErrorStateMatcher,
+              public _parentForm: NgForm,
+              public _parentFormGroup: FormGroupDirective,
+              /** INTERNAL USE ONLY. */
+              public ngControl: NgControl) {
+    super(_elementRef, _changeDetectorRef);
+  }
+}
+const _MatChipGridMixinBase: CanUpdateErrorStateCtor & typeof MatChipGridBase =
+    mixinErrorState(MatChipGridBase);
+
+/**
+ * An extension of the MatChipSet component used with MatChipRow chips and
+ * the matChipInputFor directive.
+ */
 @Component({
   moduleId: module.id,
   selector: 'mat-chip-grid',
-  templateUrl: 'chip-grid.html',
+  templateUrl: 'chip-set.html',
   styleUrls: ['chips.css'],
   host: {
-    'class': 'mat-mdc-chip-grid',
+    'class': 'mat-mdc-chip-set mat-mdc-chip-grid',
+    'role': 'grid',
+    '[attr.tabindex]': '-1',
+    '[attr.aria-describedby]': '_ariaDescribedby || null',
```

Comment:
This made me realize that this `describedby` binding could be done better on the existing `MatChipList` now that we have the `AriaDescriber`. Could you add a `TODO` here that says
```ts
// TODO: replace this binding with use of AriaDescriber
```
(not necessarily for you to take care of, just so that we keep track of it)